backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/GraphqlApi.java

```diff
@@ -91,14 +85,16 @@ public static void handleSchemaRequests(Context ctx) throws IOException {
       throw new GraphqlException(
           "Schema '" + schemaName + "' unknown. Might you need to sign in or ask permission?");
     }
-    GraphQL graphqlForSchema = session.getGraphqlForSchema(schemaName);
+    GraphQL graphqlForSchema = backend.getGraphqlForSchema(schemaName, ctx);
     ctx.header(CONTENT_TYPE, ACCEPT_JSON);
     ctx.json(executeQuery(graphqlForSchema, ctx));
   }
 
   private static String executeQuery(GraphQL g, Context ctx) throws IOException {
     String query = getQueryFromRequest(ctx);
     Map<String, Object> variables = getVariablesFromRequest(ctx);
+    MolgenisSessionManager sessionManager = new HttpMolgenisSessionManager(ctx.req());
+    Map graphQLContext = Map.of(MolgenisSessionManager.class, sessionManager);
```

Comment:
here we wrap current request so we can if needed create or destroy sessions on it